src/lib.rs

```diff
@@ -185,3 +213,78 @@ fn parse_sealed_impl(item_impl: &syn::ItemImpl) -> syn::Result<TokenStream2> {
         #item_impl
     })
 }
+
+/// Constructs [`syn::Ident`] of a sealing module name.
+fn seal_name<D: fmt::Display>(seal: D) -> syn::Ident {
+    format_ident!("__seal_{}", &seal.to_string().to_snake_case())
+}
+
+/// Arguments accepted by `#[sealed]` attribute when placed on a trait
+/// definition.
+struct TraitArguments {
+    /// `erase` argument indicating whether trait bounds erasure should be used.
+    ///
+    /// Default is `false`.
+    erased: bool,
+
+    /// `pub` argument defining visibility of the generated sealing module.
+    ///
+    /// Default is [`syn::Visibility::Inherited`].
+    visibility: syn::Visibility,
+}
+
+impl Default for TraitArguments {
+    fn default() -> Self {
+        Self {
+            erased: false,
+            visibility: syn::Visibility::Inherited,
+        }
+    }
+}
+
+impl Parse for TraitArguments {
+    fn parse(input: ParseStream<'_>) -> syn::Result<Self> {
+        let mut out = Self::default();
+
+        while !input.is_empty() {
+            let ident = syn::Ident::parse_any(&input.fork())?;
+
+            match ident.to_string().as_str() {
+                "erase" => {
+                    syn::Ident::parse_any(input)?;
+                    out.erased = true;
+                }
+
+                "pub" => {
+                    out.visibility = input.parse()?;
+                    if matches!(out.visibility, syn::Visibility::Public(_)) {
+                        return Err(syn::Error::new(
+                            out.visibility.span(),
+                            "seal with `pub` visibility is nonsense",
```

Comment:
Provide a reason as to why it is "nonsense" instead

Reply:
@jmg-duarte is something like 
```
`pub` visibility cannot seal anything
```
will be acceptable enough?

Reply:
What about
```
`pub` visibility is not strict enough. Consider refining the visibility level to something stricter (e.g. `pub(crate)`) or if you actually need sealing.
```
Possibly, we could add a link to the API Guidelines book?

Reply:
@jmg-duarte hmmm... something like this? 

```
error: `pub` visibility allows to use the seal outside its crate.
Consider tightening the visibility level (e.g. `pub(crate)`) if you actually need sealing.
Read more here: https://rust-lang.github.io/api-guidelines/future-proofing.html#sealed-traits-protect-against-downstream-implementations-c-sealed
```

Reply:
Yes! But the link may be too much, now that I see it rendered

What are your thoughts on it?

Reply:
@jmg-duarte I agree to go without the link.

Reply:
Then I think you just need to push the changes and hopefully the CI won't raise problems with the UI tests